Makefile

```diff
@@ -1,123 +1,134 @@
 CLONE_URL=github.com/pilosa/pilosa
-PKGS := $(shell cd $(GOPATH)/src/$(CLONE_URL); go list ./... | grep -v vendor)
-BUILD_TIME=`date -u +%FT%T%z`
+VERSION := $(shell git describe --tags 2> /dev/null || echo unknown)
+VERSION_ID := $(VERSION)-$(GOOS)-$(GOARCH)
+BRANCH := $(if $(TRAVIS_BRANCH),$(TRAVIS_BRANCH),$(shell git rev-parse --abbrev-ref HEAD))
+BRANCH_ID := $(BRANCH)-$(GOOS)-$(GOARCH)
+BUILD_TIME := $(shell date -u +%FT%T%z)
 LDFLAGS="-X github.com/pilosa/pilosa.Version=$(VERSION) -X github.com/pilosa/pilosa.BuildTime=$(BUILD_TIME)"
-DOCKER_GOLANG_IMAGE=golang:latest
-GIT_BRANCH := $(shell git rev-parse --abbrev-ref HEAD)
-BRANCH := $(if $(TRAVIS_BRANCH),$(TRAVIS_BRANCH),$(GIT_BRANCH))
-BRANCH_IDENTIFIER := $(BRANCH)-$(GOOS)-$(GOARCH)
+GO_VERSION=latest
 
-default: test pilosa
+# Run tests and compile Pilosa
+default: test build
 
+# Remove vendor and build directories
 clean:
 	rm -rf vendor build
 
-$(GOPATH)/bin:
-	mkdir $(GOPATH)/bin
-
-dep: $(GOPATH)/bin
-	go get -u github.com/golang/dep/cmd/dep
-
+# Set up vendor directory using `dep`
 vendor: Gopkg.toml
-ifndef DEP
-	make dep
-endif
+	$(MAKE) require-dep
 	dep ensure
 	touch vendor
 
-Gopkg.lock: dep Gopkg.toml
-	dep ensure
-
+# Run test suite
 test: vendor
-	go test $(PKGS) $(TESTFLAGS)
+	go test ./... $(TESTFLAGS)
 
+# Run test suite with coverage enabled
 cover: vendor
-	mkdir -p build/coverage
-	echo "mode: set" > build/coverage/all.out
-	for pkg in $(PKGS) ; do \
-		make cover-pkg PKG=$$pkg ; \
-	done
-
-cover-pkg:
-	mkdir -p build/coverage
-	touch build/coverage/$(subst /,-,$(PKG)).out
-	go test -coverprofile=build/coverage/$(subst /,-,$(PKG)).out $(PKG)
-	tail -n +2 build/coverage/$(subst /,-,$(PKG)).out >> build/coverage/all.out
+	mkdir -p build
+	$(MAKE) test TESTFLAGS="-coverprofile=build/coverage.out"
 
+# Run test suite with coverage enabled and view coverage results in browser
 cover-viz: cover
-	go tool cover -html=build/coverage/all.out
+	go tool cover -html=build/coverage.out
 
-pilosa: vendor
-	go build -tags release -ldflags $(LDFLAGS) $(FLAGS) $(CLONE_URL)/cmd/pilosa
+# Compile Pilosa
+build: vendor
+	go build -tags release -ldflags $(LDFLAGS) $(FLAGS) ./cmd/pilosa
 
+# Create a single release build under the build directory
 release-build: vendor
-ifdef DOCKER_BUILD
-	make docker-build FLAGS="-o build/pilosa-$(IDENTIFIER)/pilosa"
-else
-	make pilosa FLAGS="-o build/pilosa-$(IDENTIFIER)/pilosa"
-endif
-	cp LICENSE README.md build/pilosa-$(IDENTIFIER)
-	tar -cvz -C build -f build/pilosa-$(IDENTIFIER).tar.gz pilosa-$(IDENTIFIER)/
-	@echo "Created release build: build/pilosa-$(IDENTIFIER).tar.gz"
-
-release:
-ifeq ($(STATUS),"")
-	make release-build GOOS=darwin GOARCH=amd64
-	make release-build GOOS=linux GOARCH=amd64 DOCKER_BUILD=1
-	make release-build GOOS=linux GOARCH=386 DOCKER_BUILD=1
-else
-	@echo "Will not create release with unclean git status."
-endif
-
+	$(MAKE) $(if DOCKER_BUILD,docker-)build FLAGS="-o build/pilosa-$(VERSION_ID)/pilosa"
```

Comment:
looks like there should also be a `mkdir` in here and `release-build`

Reply:
I've tested that the go build process will create the directory automatically, so I think it's only needed in that one place.